modules/apps/product-navigation/product-navigation-taglib/src/main/resources/META-INF/resources/personal_menu/js/PersonalMenu.es.js

```diff
@@ -22,30 +22,31 @@ import React, {useRef, useState} from 'react';
 
 function mapItemsOnClick(items) {
 	return items.map((item) => {
-		if (item.items) {
-			item.items = [...mapItemsOnClick(item.items)];
+		const {items: nestedItems, jsOnClickConfig, ...otherKeys} = item;
+
+		const newVal = {...otherKeys};
```

Comment:
The reason I added that is to make sure we are cloning the item rather since we are mutating it by adding `.items` and `.onClick` below.

We could get around this, but we would have to spread it again into a new object at some point, to avoid mutations.

And just to note, mutations aren't all that bad, and might be more performant, but the reason we generally try to avoid them is because those are often hard to catch bugs in JS and the react render lifecycle. So I typically opt for explicit cloning and not mutating. In areas of concern for performance, we do mutate though. This should only be run once though, since the items are only fetched once.